src/web/src/components/CourseListing.vue

```diff
@@ -1,7 +1,14 @@
 <template>
   <div>
     <div class="d-flex w-100 justify-content-between">
-      <div>
+      <div v-if="actions.add" @click="toggleCourse(course)">
```

Comment:
what about `@click="actions.add ? toggleCourse(course) : toggleShowSection()"`
or move it to another function like `defaultToggleAction`